components/script/dom/range.rs

```diff
@@ -1252,9 +1251,8 @@ impl WeakRangeVec {
                 let move_start = node_is_start && start_offset > offset;
                 let move_end = node_is_end && end_offset > offset;
 
-                let remove_from_node = move_start && move_end ||
-                    move_start && !node_is_end ||
-                    move_end && !node_is_start;
+                let remove_from_node =
+                    !(!move_start || !move_end && node_is_end) || move_end && !node_is_start;
```

Comment:
I feel the same way about this new code.

Reply:
Let me work on it sir